test/elixir_internal_certificate/account/accounts_test.exs

```diff
@@ -0,0 +1,204 @@
+
+  alias ElixirInternalCertificate.Account.Accounts
+  alias ElixirInternalCertificate.Account.Schemas.{User, UserToken}
+
+  describe "get_user_by_email/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when the email does not exist, does not return the user" do
+      assert Accounts.get_user_by_email("unknown@example.com") == nil
+    end
+
+    test "with the existed email, returns the user", %{user: user} do
+      %{id: id} = user
+
+      assert %User{id: ^id} = Accounts.get_user_by_email(user.email)
+    end
+  end
+
+  describe "get_user_by_email_and_password/2" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when the email does not exist, does not return the user" do
+      assert Accounts.get_user_by_email_and_password("unknown@example.com", "hello world!") == nil
+    end
+
+    test "when the password is not valid, does not return the user", %{user: user} do
+      assert Accounts.get_user_by_email_and_password(user.email, "invalid") == nil
+    end
+
+    test "when the email and password are valid, returns the user", %{user: user} do
+      %{id: id} = user
+
+      assert %User{id: ^id} =
+               Accounts.get_user_by_email_and_password(user.email, valid_user_password())
+    end
+  end
+
+  describe "get_user!/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when id is invalid, raises a no result error" do
+      assert_raise Ecto.NoResultsError, fn ->
+        Accounts.get_user!(-1)
+      end
+    end
+
+    test "with the given id, returns the user", %{user: user} do
+      %{id: id} = user
+
+      assert %User{id: ^id} = Accounts.get_user!(user.id)
+    end
+  end
+
+  describe "register_user/1" do
+    setup do
+      %{user: insert(:user)}
+    end
+
+    test "when email and password are blank, requires validation error will be throw" do
+      {:error, changeset} = Accounts.register_user(%{})
+
+      assert %{
+               password: ["can't be blank"],
+               email: ["can't be blank"]
+             } = errors_on(changeset)
+    end
+
+    test "when email and password are not valid, requires validation error will be throw" do
+      {:error, changeset} = Accounts.register_user(%{email: "not valid", password: "short"})
+
+      assert %{
+               email: ["must have the @ sign and no spaces"],
+               password: ["should be at least 6 character(s)"]
+             } = errors_on(changeset)
+    end
+
+    test "when email and password are too long, requires validation error will be throw" do
+      too_long = String.duplicate("db", 100)
+
+      {:error, changeset} = Accounts.register_user(%{email: too_long, password: too_long})
+
+      assert "should be at most 160 character(s)" in errors_on(changeset).email
+      assert "should be at most 72 character(s)" in errors_on(changeset).password
+    end
+
+    test "when email is already existed, requires validation error will be throw", %{user: user} do
```

Comment:
we only need the user in the database only in this test, the other tests in this `describe "register_user/1" do` scope don't need that, is it worth to use the `setup` hook there @nvminhtue